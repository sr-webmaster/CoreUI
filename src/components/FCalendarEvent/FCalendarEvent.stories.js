import { MAIN } from '../../../.storybook/categories'
import { action } from '@storybook/addon-actions'
import FCalendarEvent from './FCalendarEvent.vue'
import FBtnMenu from '../FBtnMenu.vue'

const items = [{
  id: 'item1',
  label: 'Item 1'
}, {
  id: 'item2',
  label: 'Item 2'
}]

export default {
  title: `${MAIN}|FCalendarEvent`,
  id: 'FCalendarEvent',
  decorators: [
    () => ({
      template: `
        <v-container><story /></v-container>
      `
    })
  ]
}

/**
 * https://storybook.js.org/docs/guides/guide-vue/#step-4-write-your-stories
 */
export const defaultStory = () => ({
  components: {
    FCalendarEvent
  },
  data: () => ({
    date: new Date()
  }),
  template: `
    <f-calendar-event 
      title="This event is happening today, right now."
      :date="date"
    />
  `
})

export const dateAsString = () => ({
  components: {
    FCalendarEvent
  },
  data: () => ({
    date: '2020-01-01 09:00'
  }),
  template: `
    <f-calendar-event 
      title="Happy New Year!"
      subtitle="Holidays"
      :date="date"
      href="/calendar/event/1"
    />
  `
})

export const withAppendSlotContent = () => ({
  components: {
    FCalendarEvent,
    FBtnMenu
  },
  data: () => ({
    date: new Date(),
    items
  }),
  template: `
    <f-calendar-event 
      title="This event is happening today, right now."
      subtitle="Mind Blown"
      :date="date"
      @click="eventClick"
    >
      <template v-slot:append>
        <f-btn-menu :items="items" />
      </template>
    </f-calendar-event>
  `,
  methods: {
    eventClick () {
      action('eventClick')()
    }
  }
})
