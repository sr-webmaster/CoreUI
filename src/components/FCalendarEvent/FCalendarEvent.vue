<template>
  <v-list-tile class="px-0">
    <v-list-tile-action
      :class="`${color}--text f-calendar-event__cal text-xs-center py-0 pr-2 mr-2 pl-0`"
    >
      <div class="primary f-calendar-event__month-day">
        <slot
          name="month"
          :dt="dt"
        >
          <div
            class="f-calendar-event__month"
          >
            {{ monthAbbr }}
          </div>
        </slot>

        <slot
          name="date"
          :dt="dt"
        >
          <div
            class="f-calendar-event__date headline"
          >
            {{ dt.getDate() }}
          </div>
        </slot>
      </div>
      <slot
        name="time"
        :dt="dt"
      >
        <div
          class="f-calendar-event__time secondary caption pa-1"
        >
          {{ time }}
        </div>
      </slot>
    </v-list-tile-action>

    <v-list-tile-content>
      <v-list-tile-title>
        <slot
          name="title"
          :title="title"
        >
          <div
            class="f-calendar-event__title subtitle"
          >
            <f-title-link
              v-if="useTitleLink"
              v-bind="$attrs"
              v-on="$listeners"
            >
              {{ title }}
            </f-title-link>
            <span
              v-else
              class="primary--text"
            >
              {{ title }}
            </span>
          </div>
        </slot>
      </v-list-tile-title>
      <v-list-tile-sub-title>
        <slot
          name="subtitle"
          :subtitle="subtitle"
        >
          <div
            class="f-calendar-event__subtitle body-2 grey--text"
          >
            {{ subtitle }}
          </div>
        </slot>
      </v-list-tile-sub-title>
      <slot name="append" />
    </v-list-tile-content>
  </v-list-tile>
</template>

<script>
import FTitleLink from '@freshinup/core-ui/src/components/FTitleLink'
const monthAbbrs = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'
]

export default {
  components: {
    FTitleLink
  },
  props: {
    date: {
      type: [Date, String],
      required: true
    },
    title: {
      type: String,
      required: true
    },
    useTitleLink: {
      type: Boolean,
      default: false
    },
    subtitle: {
      type: String,
      default: ''
    },
    color: {
      type: String,
      default: 'white'
    }
  },
  data () {
    return {
      dt: new Date(this.date)
    }
  },
  computed: {
    monthAbbr () {
      return monthAbbrs[this.dt.getMonth()]
    },
    time () {
      return (this.dt.getHours() % 12 || 12) + ':' + ('0' + this.dt.getMinutes()).slice(-2) + this.ampm
    },
    ampm () {
      return this.dt.getHours() >= 12 ? ' PM' : ' AM'
    }
  }
}
</script>

<style scoped>
.f-calendar-event__title {
  white-space: normal;
}
.f-calendar-event__cal {
  min-width: 86px;
}
.f-calendar-event__month-day {
  width: 100%;
  border-radius: 5px 5px 0 0;
}
.f-calendar-event__time {
  width: 100%;
  border-radius: 0 0 5px 5px;
}
::v-deep .v-list__tile {
 height: auto;
 align-items: start;
}
::v-deep .v-list__tile__content {
  justify-content: start;
}
::v-deep .v-list__tile__title {
  height: auto;
  overflow: initial;
}
</style>
