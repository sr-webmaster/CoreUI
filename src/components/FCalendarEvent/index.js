import Component from './FCalendarEvent'
export default Component
