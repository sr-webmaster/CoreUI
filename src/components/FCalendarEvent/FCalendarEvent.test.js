import { mount } from '@vue/test-utils'
import createLocalVue from 'vue-cli-plugin-freshinup-ui/utils/testing/createLocalVue'
import { defaultStory, dateAsString, withAppendSlotContent } from './FCalendarEvent.stories'

describe('FCalendarEvent', () => {
  describe('Visuals', () => {
    test('defaults', () => {
      // Use mount and snapshot assertions
      const wrapper = mount(defaultStory(), {
        localVue: createLocalVue().localVue
      })

      expect(wrapper.isVueInstance()).toBe(true)
      expect(wrapper.element).toMatchSnapshot()
    })

    test('date as string', () => {
      // Use mount and snapshot assertions
      const wrapper = mount(dateAsString(), {
        localVue: createLocalVue().localVue
      })

      expect(wrapper.isVueInstance()).toBe(true)
      expect(wrapper.element).toMatchSnapshot()
    })

    test('with append slot', () => {
      // Use mount and snapshot assertions
      const wrapper = mount(withAppendSlotContent(), {
        localVue: createLocalVue().localVue
      })

      expect(wrapper.isVueInstance()).toBe(true)
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
