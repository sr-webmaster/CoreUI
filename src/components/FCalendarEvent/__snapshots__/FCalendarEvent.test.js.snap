// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FCalendarEvent Visuals date as string 1`] = `
<div
  class="px-0"
  href="/calendar/event/1"
  role="listitem"
>
  <div
    class="v-list__tile theme--light"
  >
    <div
      class="v-list__tile__action v-list__tile__action--stack white--text f-calendar-event__cal text-xs-center py-0 pr-2 mr-2 pl-0"
    >
      <div
        class="primary f-calendar-event__month-day"
      >
        <div
          class="f-calendar-event__month"
        >
          
          Jan
        
        </div>
         
        <div
          class="f-calendar-event__date headline"
        >
          
          1
        
        </div>
      </div>
       
      <div
        class="f-calendar-event__time secondary caption pa-1"
      >
        
        9:00 AM
      
      </div>
    </div>
     
    <div
      class="v-list__tile__content"
    >
      <div
        class="v-list__tile__title"
      >
        <div
          class="f-calendar-event__title subtitle"
        >
          <span
            class="primary--text"
          >
            
            Happy New Year!
          
          </span>
        </div>
      </div>
       
      <div
        class="v-list__tile__sub-title"
      >
        <div
          class="f-calendar-event__subtitle body-2 grey--text"
        >
          
          Holidays
        
        </div>
      </div>
       
    </div>
  </div>
</div>
`;

exports[`FCalendarEvent Visuals defaults 1`] = `
<div
  class="px-0"
  role="listitem"
>
  <div
    class="v-list__tile theme--light"
  >
    <div
      class="v-list__tile__action v-list__tile__action--stack white--text f-calendar-event__cal text-xs-center py-0 pr-2 mr-2 pl-0"
    >
      <div
        class="primary f-calendar-event__month-day"
      >
        <div
          class="f-calendar-event__month"
        >
          
          Jun
        
        </div>
         
        <div
          class="f-calendar-event__date headline"
        >
          
          27
        
        </div>
      </div>
       
      <div
        class="f-calendar-event__time secondary caption pa-1"
      >
        
        12:00 AM
      
      </div>
    </div>
     
    <div
      class="v-list__tile__content"
    >
      <div
        class="v-list__tile__title"
      >
        <div
          class="f-calendar-event__title subtitle"
        >
          <span
            class="primary--text"
          >
            
            This event is happening today, right now.
          
          </span>
        </div>
      </div>
       
      <div
        class="v-list__tile__sub-title"
      >
        <div
          class="f-calendar-event__subtitle body-2 grey--text"
        >
          
          
        
        </div>
      </div>
       
    </div>
  </div>
</div>
`;

exports[`FCalendarEvent Visuals with append slot 1`] = `
<div
  class="px-0"
  role="listitem"
>
  <div
    class="v-list__tile theme--light"
  >
    <div
      class="v-list__tile__action v-list__tile__action--stack white--text f-calendar-event__cal text-xs-center py-0 pr-2 mr-2 pl-0"
    >
      <div
        class="primary f-calendar-event__month-day"
      >
        <div
          class="f-calendar-event__month"
        >
          
          Jun
        
        </div>
         
        <div
          class="f-calendar-event__date headline"
        >
          
          27
        
        </div>
      </div>
       
      <div
        class="f-calendar-event__time secondary caption pa-1"
      >
        
        12:00 AM
      
      </div>
    </div>
     
    <div
      class="v-list__tile__content"
    >
      <div
        class="v-list__tile__title"
      >
        <div
          class="f-calendar-event__title subtitle"
        >
          <span
            class="primary--text"
          >
            
            This event is happening today, right now.
          
          </span>
        </div>
      </div>
       
      <div
        class="v-list__tile__sub-title"
      >
        <div
          class="f-calendar-event__subtitle body-2 grey--text"
        >
          
          Mind Blown
        
        </div>
      </div>
       
      <div
        class="v-menu"
      >
        <button
          class="v-btn theme--dark primary primary-text--text"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            Menu 
            <i
              aria-hidden="true"
              class="v-icon v-icon--right material-icons theme--dark"
            >
              arrow_drop_down
            </i>
          </div>
        </button>
        <div
          class="v-menu__content theme--light "
          style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
        >
           
          <div
            class="v-list theme--light"
            role="list"
          >
            <div
              role="listitem"
            >
              <a
                class="v-list__tile v-list__tile--link theme--light"
              >
                <div
                  class="v-list__tile__title"
                >
                  Item 1
                </div>
              </a>
            </div>
            <div
              role="listitem"
            >
              <a
                class="v-list__tile v-list__tile--link theme--light"
              >
                <div
                  class="v-list__tile__title"
                >
                  Item 2
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
